ambry-server/src/main/java/com.github.ambry.server/AmbryRequests.java

```diff
@@ -792,4 +868,33 @@ private ServerErrorCode validateRequest(PartitionId partition, RequestOrResponse
     }
     return ServerErrorCode.No_Error;
   }
+
+  /**
+   * Provides catch up status of all the remote replicas of {@code partitionIds}.
+   * @param partitionIds the {@link PartitionId}s for which lag has to be <= {@code acceptableLagInBytes}.
+   * @param acceptableLagInBytes the maximum lag in bytes that is considered "acceptable".
+   * @return {@code true} if the lag of each of the remote replicas of each of the {@link PartitionId} in
+   * {@code partitionIds} <= {@code acceptableLagInBytes}. {@code false} otherwise.
+   */
+  private boolean isAllRemoteLagLesserOrEqual(List<? extends PartitionId> partitionIds, long acceptableLagInBytes) {
+    boolean isAcceptable = true;
+    for (PartitionId partitionId : partitionIds) {
+      List<? extends ReplicaId> replicaIds = partitionId.getReplicaIds();
+      for (ReplicaId replicaId : replicaIds) {
+        if (!replicaId.getDataNodeId().equals(currentNode)) {
+          long lagInBytes = replicationManager.getRemoteReplicaLagFromLocalInBytes(partitionId,
+              replicaId.getDataNodeId().getHostname(), replicaId.getReplicaPath());
+          logger.debug("Lag of {} is {}", replicaId, lagInBytes);
+          if (lagInBytes > acceptableLagInBytes) {
+            isAcceptable = false;
+            break;
+          }
+        }
+      }
+      if (!isAcceptable) {
```

Comment:
should we move this if block to line 892 (within if block in for loop) ?

Reply:
Could you elaborate? This code is specifically to break out of the outer `for` loop and I think the only place this can be is here - something I'm missing?

Reply:
will sync up with you in person.

Reply:
sorry. I thought there is no break at 890. We are good.